### DISCUSSION TITLE
Feature request: add an option to generate new UIDs for series/study

### DISCUSSION THREAD

**Discussion by jujar** [bettar/miele-lxiv#132] [General]
OsiriX Pro/MD has a context menu option to generate a new Study instance UID, Series instance UID and/or SOP instance UID for a series or a study without anonymizing the study.  This is a feature I'd greatly appreciate in Miele-LXIV as well.

**Comment by bettar** [comment]
Please provide screenshots and a detailed description of the behaviour, including secondary effects and all conditions and implications. Don't forget also related items in the Preferences pages, if  any.

Another possibility would be to write a plugin that does whatever you like.

**Comment by jujar** [comment]
I'm sorry it has taken this long to provide additional information

- One or more studies can be selected and modified
- Clicking the context menu item will open a new dialog which will allow the user to choose which UIDs will be regenerated and if patient identity  should be unified across multiple studies and whether study should be unified as one.
- Perusing the Osirix MD preferences, I could not find anything affecting this feature.

![Screenshot 2024-09-24 at 14 04 17](https://github.com/user-attachments/assets/f579194a-80d7-492b-b654-fe714cab6f81)
![Screenshot 2024-09-24 at 14 05 18](https://github.com/user-attachments/assets/2ce864c9-729e-4253-aef6-c4a94477c4bd)
